pywbem_mock/_wbemconnection_mock.py

```diff
@@ -2913,9 +2917,21 @@ def _fake_invokemethod(self, methodname, objectname, Params, **params):
         namespace = localobject.namespace
         try:
             methodsrepo = self._get_methods_repo(namespace)
+        except CIMError as ce:
+            # If invoke indicates method repo not built, create it.
+            if ce.status_code == CIM_ERR_INVALID_NAMESPACE:
+                if namespace in self.classes or \
+                        namespace in self.instances:
+                    raise CIMError(CIM_ERR_NOT_SUPPORTED,
+                                   'Method %s in namespace %s not registered '
+                                   'in repository' %
+                                   (methodname, namespace))
+                else:
+                    raise
         except KeyError:
-            raise CIMError(CIM_ERR_NOT_FOUND, 'Method %s in namespace %s not '
-                                              'registered in repository' %
+            raise CIMError(CIM_ERR_NOT_SUPPORTED,
```

Comment:
`self._get_methods_repo()` cannot raise `KeyError` as far as I can tell from the code. I know this code was there before this change, but nevertheless we should verify and simplify this.